File: TradeItIosEmsApi/TradeItUpdateLinkRequest.m
#import "TradeItUpdateLinkRequest.h"

@implementation TradeItUpdateLinkRequest

- (id)initWithUserId:(NSString *)userId
            authInfo:(TradeItAuthenticationInfo *)authInfo
              apiKey:(NSString *)apiKey {

    self = [super init];

    if (self) {
        self.id = authInfo.id;
        self.password = authInfo.password;
        self.broker = authInfo.broker;
        self.apiKey = apiKey;
        self.userId = userId;
    }

    return self;
}

@end

File: TradeItIosEmsApi/TradeItUpdateLinkRequest.swift
class TradeItUpdateLinkRequest: TradeItRequest {

    var id: String
    var password: String
    var broker: String
    var apiKey: String
    var userId: String
    
    init(userId: String, authInfo: TradeItAuthenticationInfo, apiKey: String) {
        self.userId = userId
        self.id = authInfo.id
        self.password = authInfo.password
        self.broker = authInfo.broker
        self.apiKey = apiKey
    }
}
